File: diff_drive_control/src/spiral.cpp
#include<ros/ros.h>
#include<geometry_msgs/Twist.h>

int main(int argc, char **argv)
{
    ros::init(argc, argv, "spiral_node");
    ros::NodeHandle n;
    ros::Publisher pub = n.advertise<geometry_msgs::Twist>("/cmd_vel",1000);
    ros::Rate rate(10);
    geometry_msgs::Twist vel;
    ROS_INFO("Moving the bot.");
    float r = 0.1;
    while(ros::ok())
    {
        vel.linear.x = 0.4;
        vel.angular.z = 0.4/r;
        pub.publish(vel);
        r += 0.01;
        ros::spinOnce();
        rate.sleep();
    }

    ROS_INFO("Stopping the bot."); 
    return 0;
}

File: diff_drive_control/scripts/spiral.py
#!/usr/bin/env python

import rospy
from geometry_msgs.msg import Twist

def spiral():
    rospy.init_node('spiral_node',anonymous=True)
    pub = rospy.Publisher('/cmd_vel', Twist, queue_size=10)
    velocity = Twist()
    rate = rospy.Rate(10)
    rospy.loginfo('Moving Bot in Spiral')
    radius = 0.1

    while not rospy.is_shutdown():
        velocity.linear.x = 0.4
        velocity.angular.z = 0.4/radius
        pub.publish(velocity)
        rate.sleep()
        radius += 0.01
    
    velocity.linear.x = 0
    velocity.angular.z = 0
    pub.publish(velocity)

if __name__=='__main__':
    try:
        spiral()
    except rospy.ROSInterruptException:
        pass